workers/unity/Assets/Gdk/Core/Error/LogEvent.cs

```diff
@@ -0,0 +1,45 @@
+using System.Collections.Generic;
+using System.Text;
+using UnityEngine;
+
+namespace Improbable.Gdk.Core
+{
+    public struct LogEvent
+    {
+        public string Message;
+        public Dictionary<string, object> Data;
+        public LogType Type;
+
+        public LogEvent(string message, LogType type)
+        {
+            Message = message;
+            Type = type;
+            Data = new Dictionary<string, object>();
+        }
+
+        public LogEvent WithField(string key, object value)
+        {
+            Data.Add(key, value);
+            return this;
+        }
+
+        public override string ToString()
+        {
+            var builder = new StringBuilder();
+
+            builder.AppendLine(Message);
+
+            if (Data.Count > 0)
+            {
+                builder.AppendLine("Log event context:");
+
+                foreach (var entry in Data)
+                {
+                    builder.AppendLine($"{entry.Key}: {entry.Value}");
```

Comment:
Put `'` around key/values to help pick the value out, and make empty string obvious.